### ISSUE TITLE
Allocating an Smd2Body on the stack is likely to cause a stack overflow

### ISSUE THREAD

**Issue by Olxinos** [Megacrafter127/C-StarMadeFileParser#36]
`typedef rawCompressedSegment smd2Body[4096];`
`typedef Bytef rawCompressedSegment[5120];`
This means a single smd2Body automatic variable will take 20MB on the stack, this is greater than the stack size of most systems.

**Comment by Olxinos** [comment]
Note: using `std::array` instead of C-style arrays should solve this problem
the `T* std::array::data()` method allowing to access directly the underlying memory range is a C++11 feature though

**Comment by Megacrafter127** [comment]
yes, however an smd2Body is not supposed to be allocated on the stack, but with malloc and similar functions, but as the type is not used in the API itself anyways, you can remove it if you think users of this API will be stupid enough to allocate that on the stack.
